client/dashboard/test-utils.tsx

```diff
@@ -0,0 +1,22 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { RouterProvider, createRouter, createRootRoute } from '@tanstack/react-router';
+import { render as testingLibraryRender } from '@testing-library/react';
+import { Suspense } from 'react';
+
+export function render( ui: React.ReactElement ) {
+	const queryClient = new QueryClient( {
+		defaultOptions: {
+			queries: { retry: false },
+		},
+	} );
+	const router = createRouter( {
+		routeTree: createRootRoute( { component: () => ui } ),
+	} );
+	return testingLibraryRender(
+		<QueryClientProvider client={ queryClient }>
+			<Suspense fallback={ <div>Loading...</div> }>
```

Comment:
Doesn't the `RouterProvider` render the Suspense by default 🤔

Reply:
I think it's so that we can distinguish loading states in the tests easily 😄

Reply:
Yeah, I want something concrete to test for. Actually it might be better to use a `data-testid` on this element since we know we're testing for such a specific thing.

Afaik there isn't another way to test for "the component is no longer suspended"